src/CachePeers.h

```diff
@@ -13,8 +13,51 @@
 #include "CachePeer.h"
 #include "mem/PoolingAllocator.h"
 
+#include <memory>
 #include <vector>
 
+/// cache_peer configuration storage
+class CachePeers
+{
+public:
+    /// owns stored CachePeer objects
+    using Storage = std::vector< std::unique_ptr<CachePeer>, PoolingAllocator< std::unique_ptr<CachePeer> > >;
+
+    /// stores a being-configured cache_peer
+    void add(CachePeer *p) { storage.emplace_back(p); }
+
+    /// deletes a previously add()ed CachePeer object
+    void remove(CachePeer *);
+
+    /// the number of currently stored (i.e. added and not removed) cache_peers
+    size_t size() const { return storage.size(); }
+
+    /* peer iterators forming a sequence for C++ range-based for loop API */
+    using const_iterator = Storage::const_iterator;
+    const_iterator begin() const { return storage.cbegin(); }
+    const_iterator end() const { return storage.cend(); }
```

Comment:
I am not against this improvement, but I could not find existing examples with return type deduction in Squid code. Are we going to use this new style here (and in similar future cases)?

Reply:
> Are we going to use this new style here (and in similar future cases)?

Ideally, we should use `auto` where using it is better than not using it. Hopefully, we can avoid an endless stream of change requests asking folks to add or remove `auto` or arguing about special/tricky cases like this one.

My recommendation is to use `auto` here just to avoid arguing about it.

P.S. IIIRC, a recent PR used return type deduction; it is probably not merged yet.

Reply:
Done, I also adjusted size(), just in case.